configure.py

```diff
@@ -3555,7 +3548,7 @@ def main(argv):
 
     setup_logging(options)
 
-    source_paths = SourcePaths(os.path.dirname(argv[0]))
+    source_paths = SourcePaths(os.path.dirname(os.path.realpath(__file__)))
```

Comment:
This caused the explained issue. `argv[0]` may be a valid relative path (seen from `$($pwd)`), but it is not valid when `--with-build-dir=` drops the generated build files somewhere else and paths in there are relative to whatever `argv[0]` was.